### PR TITLE
Define value for all constant_keyword fields in apmpackage

### LINKED ISSUE #12159: Define value for all constant-keyword fields in the apm package
The apm package defines fields of type `constant_keyword` in several places. Some of these fields have a fixed value set already in the template definition; others do not. This is generally supported by ES, but causes problems in combination with how Fleet handles mapping updates and rollovers, see details in https://github.com/elastic/elasticsearch/issues/102829. Solving this in Fleet would add undesired complexity, therefore it was suggested to instead require more restrictive behaviour via the package spec, by requiring a value to be defined for constant keyword types (https://github.com/elastic/package-spec/issues/677).

We need to update the apm-package to define values for all of these fields. This includes changing the package generation logic for the interval based templates (https://github.com/elastic/apm-server/blob/main/apmpackage/cmd/genpackage/main.go#L62).

### PR DESCRIPTION
**Pull request by carsonip**
## Motivation/summary

Define value for all `constant_keyword` fields in apmpackage to workaround Fleet upgrade bug. Under certain conditions, it is possible that apmpackage upgrade creates a lot of indices if constant_keyword does not have a value.

Targeting 8.12 branch because apmpackage has been moved to https://github.com/elastic/integrations/tree/main/packages/apm/data_stream on main.

## Checklist

- [ ] Update [CHANGELOG.asciidoc](https://github.com/elastic/apm-server/blob/main/CHANGELOG.asciidoc)
- [x] Update [package changelog.yml](https://github.com/elastic/apm-server/blob/main/apmpackage/apm/changelog.yml) (only if changes to `apmpackage` have been made)
- [ ] Documentation has been updated

For functional changes, consider:
- Is it observable through the addition of either **logging** or **metrics**?
- Is its use being published in **telemetry** to enable product improvement?
- Have system tests been added to avoid regression?

## How to test these changes

- Ingest APM data, upgrade package to a version that contains the fix, and verify that DS rollover is successful and mapping contains a value for constant keyword fields.

## Related issues

Closes #12159

### CODE REVIEW

File: apmpackage/apm/data_stream/sampled_traces/fields/base-fields.yml [hunk 1]
```diff
@@ -1,12 +1,8 @@
 - name: '@timestamp'
-  type: date
-  description: Event timestamp.
+  external: ecs
```
Comment (carsonip) [inline apmpackage/apm/data_stream/sampled_traces/fields/base-fields.yml:2]: [For reviewer] This is similar to #6714

**Comment by carsonip** [comment]
@simitt now that apmpackage no longer lives in apm-server repo, should I split this PR into 2 parts, 1 targeting apm-server 8.12 branch, another changing https://github.com/elastic/integrations/tree/main/packages/apm/data_stream ?

**Comment by simitt** [comment]
Yeah I fear we need to put this into both places now unfortunately.

**Review by carsonip — COMMENTED** [review]
Once this is merged I'll port the change to https://github.com/elastic/integrations/tree/main/packages/apm

**Comment by bmorelli25** [comment]
run docs-build

File: apmpackage/apm/data_stream/sampled_traces/fields/base-fields.yml (hunk 1 above)
Comment (carsonip) [reply apmpackage/apm/data_stream/sampled_traces/fields/base-fields.yml:2]: the rendered index template is identical. So there is nothing to worry about.

**Comment by carsonip** [comment]
This change did not go into 8.12.1. Now the plan is to land this fix in 8.12.2.

**Comment by carsonip** [comment]
## testing notes

Upgraded from a 8.12.1 deployment with existing data in metrics-apm.service_destination.1m-default etc to 8.12.2. Upgrade went well and data streams rolled over successfully.

**Pull request merged**